### PR TITLE
feat(Checkbox/react): add component

### PR DESCRIPTION
**Pull request by haquezameer**

### CODE REVIEW

File: src/atoms/Checkbox/Checkbox.web.js [hunk 1]
```diff
@@ -1,3 +1,272 @@
...
+      medium: '24px',
+      small: '20px',
+      xsmall: '16px',
+    },
+    borderRadius: {
+      large: '14px',
+      medium: '12px',
+      small: '10px',
+      xsmall: '8px',
+    },
```
Comment (haquezameer) [inline src/atoms/Checkbox/Checkbox.web.js:119]: will use makePX

File: src/atoms/Checkbox/Checkbox.web.js [hunk 2]
```diff
@@ -1,3 +1,275 @@
...
+  id,
+}) => {
+  if (isDefined(defaultChecked) && isDefined(externalChecked)) {
+    throw Error('One of defaultChecked or checked should be supplied.');
+  }
+
+  const [isChecked, setIsChecked] = useState(defaultChecked || false);
+
+  const onCheckChange = useCallback(() => {
+    if (isDefined(externalChecked)) {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:152]:
```suggestion
    if (externalChecked) {
```

File: src/atoms/Checkbox/Checkbox.web.js [hunk 3]
```diff
@@ -1,3 +1,275 @@
...
+      onChange(!externalChecked);
+      return;
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:154]: Why existing early when `externalChecked` is defined?

File: src/atoms/Checkbox/Input.web.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+const Input = styled.input.attrs({
+  type: 'checkbox',
+})`
+  position: absolute;
+  width: 0px;
+  height: 0px;
+  padding: 0;
+  margin: 0;
+  overflow: hidden;
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Input.web.js:12]: Why is this needed?

File: src/atoms/Checkbox/Checkbox.web.js [hunk 5]
```diff
@@ -1,3 +1,275 @@
...
+          return 'xsmall';
+        default:
+          return 'small';
+      }
+    },
+    color({ disabled, errorText }) {
+      if (errorText) {
+        return 'negative.900';
+      }
+      if (disabled) {
```
[the remark this answers is not in the document]
Comment (haquezameer) [reply src/atoms/Checkbox/Checkbox.web.js:70]: error always overrides the disabled state. had confirmed with design.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 6]
```diff
@@ -1,3 +1,275 @@
...
+import Label from './Label';
+import Input from './Input';
+
+const styles = {
+  icon: {
+    fill({ disabled, isChecked, variantColor, externalChecked }) {
+      if (disabled) {
+        return 'shade.930';
+      }
+      if (isChecked || externalChecked) {
```
[the remark this answers is not in the document]
Comment (haquezameer) [reply src/atoms/Checkbox/Checkbox.web.js:23]: yep

File: src/atoms/Checkbox/Checkbox.web.js [hunk 7]
```diff
@@ -1,3 +1,275 @@
...
+      }
+    },
+    size(checkboxSize) {
+      switch (checkboxSize) {
+        case 'large':
+          return 'medium';
+        case 'medium':
+          return 'xsmall';
+        default:
+          return 'small';
```
[the remark this answers is not in the document]
Comment (haquezameer) [reply src/atoms/Checkbox/Checkbox.web.js:63]: yes, as per design

File: src/atoms/Checkbox/Input.web.js (hunk 4 above)
Comment (haquezameer) [reply src/atoms/Checkbox/Input.web.js:12]: This was added for clip.
> Note: The clip property does not work if "overflow:visible".
https://www.w3schools.com/cssref/pr_pos_clip.asp

clip-path is not supported in IE. clip is supported although it is deprecated. should we remove the usage of clip or keep it for safety?

File: src/atoms/Checkbox/Checkbox.web.js (hunk 3 above)
Comment (haquezameer) [reply src/atoms/Checkbox/Checkbox.web.js]: for not managing state in the component as `externalChecked` is meant for un-controlled comp.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 8]
```diff
@@ -1,3 +1,282 @@
...
+      if (disabled) {
+        return 'transparent';
+      }
+
+      switch (state) {
+        case 'hover':
+          if (!(isChecked || externalChecked)) {
+            return theme.colors.tone[930];
+          }
+          return getColor(theme, `${variantColor}.920`);
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:87]:
```suggestion
          if (isChecked || externalChecked) {
            return getColor(theme, `${variantColor}.920`);
          }
          return theme.colors.tone[930];
```

Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js:87]: Check other way around, easier to read.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 9]
```diff
@@ -1,3 +1,282 @@
...
+        case 'focus':
+          if (!(isChecked || externalChecked)) {
+            return theme.colors.tone[940];
+          }
+          return getColor(theme, `${variantColor}.930`);
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:92]: Invert check.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 10]
```diff
@@ -1,3 +1,282 @@
...
+        case 'active':
+          if (!(isChecked || externalChecked)) {
+            return theme.colors.tone[940];
+          }
+          return getColor(theme, `${variantColor}.940`);
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:97]: Invert check.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 11]
```diff
@@ -1,3 +1,282 @@
...
+        default:
+          return 'transparent';
+      }
+    },
+    width: {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:102]: Let's refactor this based on our discussion.

Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js:102]:
```js
 dimensions(size) {
      switch (size) {
        case 'large':
          return {
            width: makePxValue(..),
            height: ...,
            borderRadius: ...,
          };
```

File: src/atoms/Checkbox/Checkbox.web.js [hunk 12]
```diff
@@ -1,3 +1,282 @@
...
+  if (isDefined(defaultChecked) && isDefined(externalChecked)) {
+    throw Error('One of defaultChecked or checked should be supplied.');
+  }
+
+  const [isChecked, setIsChecked] = useState(defaultChecked || false);
+
+  const onCheckChange = useCallback(
+    (event) => {
+      /* prevent default behaviour of label firing click event on the associated input */
+      event.preventDefault();
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:154]: Leave new line after this.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 13]
```diff
@@ -1,3 +1,282 @@
...
+    },
+    [externalChecked, onChange, disabled],
+  );
+
+  const descriptionText = errorText || helpText;
+
+  return (
+    <Flex alignSelf="flex-start" flexDirection="column">
+      <View>
+        <Flex flexDirection="row" alignItems="center">
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:174]:
```suggestion
        <Flex alignItems="center">
```

Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js:174]: `row` is default value.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 14]
```diff
@@ -1,3 +1,282 @@
...
+        });
+      }
+    },
+    [externalChecked, onChange, disabled],
+  );
+
+  const descriptionText = errorText || helpText;
+
+  return (
+    <Flex alignSelf="flex-start" flexDirection="column">
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:172]:
```suggestion
    <Flex flexDirection="column" alignSelf="flex-start">
```

Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js:172]: Easier to read.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 15]
```diff
@@ -1,3 +1,282 @@
...
+            <Input
+              id={id}
+              name={name}
+              defaultChecked={isChecked || externalChecked}
+              isChecked={isChecked}
+              externalChecked={externalChecked}
+              disabled={disabled}
+              variantColor={variantColor}
+              backdropStyles={styles.backdrop}
+              {...automation(testID)}
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:185]: Move this to label.

Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js:185]: Check if tests are working with this, otherwise move.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 16]
```diff
@@ -1,3 +1,282 @@
...
+                  size={styles.icon.size({ size })}
+                  name={externalChecked ?? isChecked ? 'checkboxFilled' : 'checkboxOutlined'}
+                  fill={styles.icon.fill({ isChecked, disabled, variantColor, externalChecked })}
+                />
+              </Backdrop>
+            </Size>
+            <View>
+              {title ? (
+                <Flex>
+                  <Space margin={[0, 0, 0, 0.5]}>
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:199]: Move styles to `styles.title`.

Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:199]:
```suggestion
                  <Space margin={styles.title.margin()}>
```

File: src/atoms/Checkbox/Checkbox.web.js [hunk 17]
```diff
@@ -1,3 +1,282 @@
...
+            <Size width={styles.backdrop.width[size]} height={styles.backdrop.height[size]}>
+              <Backdrop borderRadius={styles.backdrop.borderRadius[size]}>
+                <Icon
+                  size={styles.icon.size({ size })}
+                  name={externalChecked ?? isChecked ? 'checkboxFilled' : 'checkboxOutlined'}
+                  fill={styles.icon.fill({ isChecked, disabled, variantColor, externalChecked })}
+                />
+              </Backdrop>
+            </Size>
+            <View>
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:196]: Is this really needed?

File: src/atoms/Checkbox/Checkbox.web.js [hunk 18]
```diff
@@ -1,3 +1,282 @@
...
+              {title ? (
+                <Flex>
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:198]: Is this really needed?

File: src/atoms/Checkbox/Checkbox.web.js [hunk 19]
```diff
@@ -1,3 +1,282 @@
...
+                  <Space margin={[0, 0, 0, 0.5]}>
+                    <View>
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:200]: Is this really needed?

File: src/atoms/Checkbox/Checkbox.web.js [hunk 20]
```diff
@@ -1,3 +1,282 @@
...
+                      </Text>
+                    </View>
+                  </Space>
+                </Flex>
+              ) : null}
+            </View>
+          </Label>
+        </Flex>
+
+        {title && (!isEmpty(helpText) || !isEmpty(errorText)) && size !== 'small' ? (
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:212]: Move this check to a variable `shouldShowDescriptionText`

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 21]
```diff
@@ -0,0 +1,204 @@
...
+import Checkbox from '../index';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+const SAMPLE_ID = 'sample-checkbox';
+
+describe('<Checkbox />', () => {
+  describe('id', () => {
+    it('should not have id attribute if not provded as prop', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:13]:
```suggestion
    it('should not have id attribute if not provided as prop', () => {
```

Comment (dhruvdutt) [reply src/atoms/Checkbox/__tests__/Checkbox.web.test.js:13]: Fix this everywhere.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 22]
```diff
@@ -0,0 +1,204 @@
...
+      const { getByRole } = renderWithTheme(<Checkbox defaultChecked={false} />);
+      const checkbox = getByRole('checkbox');
+      expect(checkbox).not.toHaveAttribute('id');
+    });
+
+    it('should have provided id attribute if id provded as prop', () => {
+      const { getByRole } = renderWithTheme(<Checkbox id={SAMPLE_ID} defaultChecked={false} />);
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:20]:
```suggestion
      const { getByRole } = renderWithTheme(<Checkbox id={SAMPLE_ID} />);
```

Comment (dhruvdutt) [reply src/atoms/Checkbox/__tests__/Checkbox.web.test.js:20]: Update the same in all tests.

File: src/atoms/Checkbox/Checkbox.web.js [hunk 23]
```diff
@@ -1,3 +1,282 @@
...
+};
+
+Checkbox.defaultProps = {
+  size: 'medium',
+  helpText: '',
+  disabled: false,
+  errorText: '',
+  variantColor: 'primary',
+  testID: 'ds-checkbox',
+  onChange: () => {},
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Checkbox.web.js:279]: Add `defaultChecked` as `false`.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 24]
```diff
@@ -0,0 +1,204 @@
...
+          title="Nice checkbox"
+          errorText={errorText}
+        />,
+      );
+      expect(container).toHaveTextContent(errorText);
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('onChange', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:112]: Add test for controller, un-controlled both. With `checked` prop as well.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 25]
```diff
@@ -0,0 +1,204 @@
...
+        <Checkbox id={SAMPLE_ID} defaultChecked={false} />,
+      );
+      const checkbox = getByRole('checkbox');
+      expect(checkbox.checked).toBe(false);
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should render checkbox checked(attr)=true if defaultChecked(prop) is true', () => {
+      const { container, getByRole } = renderWithTheme(
+        <Checkbox id={SAMPLE_ID} defaultChecked={true} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:148]:
```suggestion
        <Checkbox id={SAMPLE_ID} defaultChecked />,
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 26]
```diff
@@ -0,0 +1,204 @@
...
+
+  describe('checked', () => {
+    it('should render checkbox with checked(attr)=false if checked(prop) is false', () => {
+      const { container, getByRole } = renderWithTheme(<Checkbox id={SAMPLE_ID} checked={false} />);
+      const checkbox = getByRole('checkbox');
+      expect(checkbox.checked).toBe(false);
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should render checkbox checked(attr)=true if checked(prop) is true', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:175]:
```suggestion
    it('should render checkbox checked if checked(prop) is true', () => {
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 27]
```diff
@@ -0,0 +1,204 @@
...
+        <Checkbox id={SAMPLE_ID} disabled={true} checked={false} />,
+      );
+      const checkbox = getByRole('checkbox');
+      expect(checkbox).toBeDisabled();
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('checked', () => {
+    it('should render checkbox with checked(attr)=false if checked(prop) is false', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:168]:
```suggestion
    it('should render checkbox with checked(attr) as false if checked(prop) is false', () => {
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js (hunk 26 above)
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:175]:
```suggestion
    it('should render checkbox checked(attr) as true if checked(prop) is true', () => {
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 28]
```diff
@@ -0,0 +1,204 @@
...
+    it('should render checkbox with checked(attr)=false if defaultChecked(prop) is false', () => {
+      const { container, getByRole } = renderWithTheme(
+        <Checkbox id={SAMPLE_ID} defaultChecked={false} />,
+      );
+      const checkbox = getByRole('checkbox');
+      expect(checkbox.checked).toBe(false);
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should render checkbox checked(attr)=true if defaultChecked(prop) is true', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:146]:
```suggestion
    it('should render checkbox checked(attr) as true if defaultChecked(prop) is true', () => {
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 29]
```diff
@@ -0,0 +1,204 @@
...
+        );
+        const checkBox = getByRole('checkbox');
+        fireEvent.click(checkBox);
+        expect(onChange).not.toBeCalled();
+      });
+    });
+  });
+
+  describe('defaultChecked', () => {
+    it('should render checkbox with checked(attr)=false if defaultChecked(prop) is false', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:137]:
```suggestion
    it('should render checkbox with checked(attr) as false if defaultChecked(prop) is false', () => {
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 30]
```diff
@@ -0,0 +1,204 @@
...
+      const checkbox = getByRole('checkbox');
+      expect(checkbox.checked).toBe(true);
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('disabled', () => {
+    it('should render a disabled checkbox if disabled is true', () => {
+      const { container, getByRole } = renderWithTheme(
+        <Checkbox id={SAMPLE_ID} disabled={true} checked={false} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:159]:
```suggestion
        <Checkbox id={SAMPLE_ID} disabled checked={false} />,
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 31]
```diff
@@ -0,0 +1,204 @@
...
+  describe('checked', () => {
+    it('should render checkbox with checked(attr)=false if checked(prop) is false', () => {
+      const { container, getByRole } = renderWithTheme(<Checkbox id={SAMPLE_ID} checked={false} />);
+      const checkbox = getByRole('checkbox');
+      expect(checkbox.checked).toBe(false);
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should render checkbox checked(attr)=true if checked(prop) is true', () => {
+      const { container, getByRole } = renderWithTheme(<Checkbox id={SAMPLE_ID} checked={true} />);
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:176]:
```suggestion
      const { container, getByRole } = renderWithTheme(<Checkbox id={SAMPLE_ID} checked />);
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 32]
```diff
@@ -0,0 +1,204 @@
...
+import { fireEvent } from '@testing-library/react';
+import { renderWithTheme } from '../../../_helpers/testing';
+import Checkbox from '../index';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+const SAMPLE_ID = 'sample-checkbox';
+
+describe('<Checkbox />', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:11]: Add tests for `title`.

Comment (dhruvdutt) [reply src/atoms/Checkbox/__tests__/Checkbox.web.test.js:11]: Add tests for icon only / without title.

File: src/atoms/Icon/Icon.web.js [hunk 33]
```diff
@@ -32,7 +33,7 @@ Icon.defaultProps = {
 };
 
 Icon.propTypes = {
-  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
+  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
```
Comment (dhruvdutt) [inline src/atoms/Icon/Icon.web.js:36]: Why?

File: src/atoms/Checkbox/Checkbox.web.js (hunk 15 above)
Comment (haquezameer) [reply src/atoms/Checkbox/Checkbox.web.js:207]: works

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 34]
```diff
@@ -156,7 +136,7 @@ describe('<Checkbox />', () => {
   describe('disabled', () => {
     it('should render a disabled checkbox if disabled is true', () => {
       const { container, getByRole } = renderWithTheme(
-        <Checkbox id={SAMPLE_ID} disabled={true} checked={false} />,
+        <Checkbox id={SAMPLE_ID} disabled defaultChecked={false} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:235]: Why do you need to pass `defaultChecked` if it's `false` anyway?

File: src/atoms/Checkbox/Input.web.js [hunk 35]
```diff
@@ -5,8 +5,8 @@ const Input = styled.input.attrs({
   type: 'checkbox',
 })`
   position: absolute;
-  width: 0px;
-  height: 0px;
+  width: 1px;
+  height: 1px;
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/Input.web.js:9]: Why is this needed?

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 36]
```diff
@@ -91,26 +100,113 @@ describe('<Checkbox />', () => {
     });
   });
 
+  describe('checkbox only', () => {
+    it('should render a icon only checkbox', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:104]:
```suggestion
    it('should render an icon only checkbox', () => {
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 37]
```diff
@@ -91,26 +100,113 @@ describe('<Checkbox />', () => {
...
+        );
+        const checkbox = getByRole('checkbox');
+        checkbox.focus();
+        expect(checkbox).toHaveFocus();
+        expect(container).toMatchSnapshot();
+      });
+
+      it('should render focus styles on checkbox when defaultChecked is true', () => {
+        const { container, getByRole } = renderWithTheme(
+          <Checkbox id={SAMPLE_ID} defaultChecked={true} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:124]:
```suggestion
          <Checkbox id={SAMPLE_ID} defaultChecked />,
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 38]
```diff
@@ -91,26 +100,113 @@ describe('<Checkbox />', () => {
...
+        checkbox.focus();
+        expect(checkbox).toHaveFocus();
+        expect(container).toMatchSnapshot();
+      });
+    });
+
+    describe('with checked', () => {
+      it('should render focus styles on checkbox when defaultChecked is false', () => {
+        const { container, getByRole } = renderWithTheme(
+          <Checkbox id={SAMPLE_ID} checked={false} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:136]: The test description says "defaultChecked is false"

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 39]
```diff
@@ -91,26 +100,113 @@ describe('<Checkbox />', () => {
...
+        );
+        const checkbox = getByRole('checkbox');
+        checkbox.focus();
+        expect(checkbox).toHaveFocus();
+        expect(container).toMatchSnapshot();
+      });
+
+      it('should render focus styles on checkbox when defaultChecked is true', () => {
+        const { container, getByRole } = renderWithTheme(
+          <Checkbox id={SAMPLE_ID} checked={true} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:146]: The test description says "defaultChecked is true"

Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:146]:
```suggestion
          <Checkbox id={SAMPLE_ID} checked />,
```

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 40]
```diff
@@ -91,26 +100,113 @@ describe('<Checkbox />', () => {
...
   describe('onChange', () => {
-    it('should call onChange with updated check value', () => {
-      const onChange = jest.fn();
-      const { getByRole } = renderWithTheme(
-        <Checkbox id={SAMPLE_ID} defaultChecked={false} onChange={onChange} />,
-      );
-      const checkBox = getByRole('checkbox');
-      fireEvent.click(checkBox);
-      expect(onChange).toHaveBeenCalledTimes(1);
+    describe('with defaultChecked(uncontrolled)', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:157]: For uncontrolled, add test when `defaultChecked` is true.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 41]
```diff
@@ -91,26 +100,113 @@ describe('<Checkbox />', () => {
...
+          fireEvent.click(checkbox);
+          expect(checkbox.defaultChecked).toBe(false);
+          expect(onChange).not.toBeCalled();
+        });
+      });
     });
 
-    describe('with disabled', () => {
-      it('should not call onChange if checkbox is disabled', () => {
+    describe('with checked(controlled)', () => {
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:185]: For controlled, add test when `checked` is initially true.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js [hunk 42]
```diff
@@ -141,7 +141,7 @@ describe('<Checkbox />', () => {
         expect(container).toMatchSnapshot();
       });
 
-      it('should render focus styles on checkbox when defaultChecked is true', () => {
+      it('should render focus styles on checkbox when checked is true', () => {
         const { container, getByRole } = renderWithTheme(
           <Checkbox id={SAMPLE_ID} checked={true} />,
```
Comment (dhruvdutt) [inline src/atoms/Checkbox/__tests__/Checkbox.web.test.js:146]:
```suggestion
          <Checkbox id={SAMPLE_ID} checked />,
```

File: src/atoms/Checkbox/Checkbox.web.js (hunk 23 above)
Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js:296]: Ignore this.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js (hunk 34 above)
Comment (dhruvdutt) [reply src/atoms/Checkbox/__tests__/Checkbox.web.test.js:235]: Ignore.

File: src/atoms/Checkbox/Checkbox.web.js (hunk 2 above)
Comment (dhruvdutt) [reply src/atoms/Checkbox/Checkbox.web.js]: Let it be, this will be required.

File: src/atoms/Checkbox/__tests__/Checkbox.web.test.js (hunk 39 above)
Comment (haquezameer) [reply src/atoms/Checkbox/__tests__/Checkbox.web.test.js:146]: had updated this things in last push.

File: src/atoms/Checkbox/Input.web.js (hunk 35 above)
Comment (haquezameer) [reply src/atoms/Checkbox/Input.web.js:9]: > An element with no height or width, whether defined in HTML or CSS, is typically removed from the flow of the page, so most screen readers will not read it.

https://webaim.org/techniques/css/invisiblecontent/
